src/sortStudents.ts

```diff
@@ -1,16 +1,70 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'average',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' |'desc';
 
+function getAverageGrade(grades: number[]): number {
+  const sum = grades.reduce((acc, value) => acc + value, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sum / grades.length;
+}
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        return [...students].sort(
+          (a, b) => a[sortBy].localeCompare(b[sortBy]),
+        );
+      }
+
+      return [...students].sort(
+        (a, b) => b[sortBy].localeCompare(a[sortBy]),
+      );
```

Comment:
Can be shortened, same for the rest of `cases`.
```suggestion
  const sortOrder = order === 'asc' ? 1 : -1;
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      return [...students].sort(
        (a, b) => sortOrder * a[sortBy].localeCompare(b[sortBy]),
      );
```